src/wavesurfer.js

```diff
@@ -283,17 +283,17 @@ var WaveSurfer = {
     /**
      * Internal method.
      */
-    loadArrayBuffer: function (arraybuffer) {
+    loadArrayBuffer: function (arraybuffer, peaks) {
         this.decodeArrayBuffer(arraybuffer, function (data) {
-            this.loadDecodedBuffer(data);
+            this.loadDecodedBuffer(data, peaks);
```

Comment:
I wonder if we could make it work in the same way as Media Element. I.e. draw the waveform first, and then decode the buffer.